aya/src/obj/mod.rs

```diff
@@ -510,13 +509,24 @@ fn parse_map(section: &Section, name: &str) -> Result<Map, ParseError> {
 }
 
 fn parse_map_def(name: &str, data: &[u8]) -> Result<bpf_map_def, ParseError> {
-    if mem::size_of::<bpf_map_def>() > data.len() {
+    if data.len() > mem::size_of::<bpf_map_def>() {
```

Comment:
We also need to check that it's at least as big as up to `map_flags`

Reply:
I added the check and put back the tests for a too-small definition. Now it should check for too large and too small.